src/screens/MessageScreen.tsx

```diff
@@ -18,21 +22,63 @@ export function MessageScreen({
   const { recipientsUserIds } = route.params;
   const { styles } = useStyles(defaultStyles);
   const { data, isLoading: userLoading } = useUser();
+  const { User } = useIcons();
+
+  const [scrollIndex, setScrollIndex] = useState<number>(0);
+
+  // Users that have sent new messages
+  const { userIds: unreads, isLoading } = useHasNewMessagesFromUsers({
+    currentUserId: data?.id,
+    userIds: recipientsUserIds,
+  });
+
+  // Sort recipientsList by unreads
+  const sortedList = useMemo(
+    () =>
+      orderBy(recipientsUserIds, (userId) => unreads?.includes(userId), 'desc'),
+    [unreads, recipientsUserIds],
+  );
+
+  // Break-down list of users into smaller chunks
+  const recipientsListChunked = useMemo(
+    () => chunk(sortedList, 10),
+    [sortedList],
+  );
+
+  // Use current scroll-index to expand the list of users
+  // in view up to the maximum
+  const usersInView = useMemo(
```

Comment:
Does this mean that we won't fetch the unread status until the user scrolls so that a recipient is in view?

Reply:
I think not based on this logic:
- The recipientListChunked is created using the sortedList 
- The sortedList is created by sorting the recipientsUserIds based on the unreadIds
- The unreadIds checks for unread messages from every user